File: JavaLibrary/src/test/java/com/flat/simplex/parser/logic/line/call/CallGroupTest.java
package com.flat.simplex.parser.logic.line.call;

import com.flat.simplex.parser.logic.Context;
import com.flat.simplex.parser.logic.block.BlockIf;
import com.flat.simplex.parser.logic.error.Error;
import com.flat.simplex.support.TokenChain;
import org.junit.jupiter.api.Test;

import static com.flat.simplex.support.ContextSupport.assertErrors;
import static com.flat.simplex.support.TokenChain.readChain;
import static org.junit.jupiter.api.Assertions.*;

class CallGroupTest {

    @Test
    public void loadGroup() {
        TokenChain chain = readChain("(a)");
        Context context = new Context(chain.get());
        CallGroup call = new CallGroup(context, chain.get());
        call.load();

        assertNotNull(call.getLineValue(), "Invalid line");

        assertErrors(context);
    }

    @Test
    public void loadEmpty_Fail() {
        TokenChain chain = readChain("()");
        Context context = new Context(chain.get());
        CallGroup call = new CallGroup(context, chain.get());
        call.load();

        assertNull(call.getLineValue(), "Invalid line");

        assertErrors(context, Error.lineEmptyLine);
    }

    @Test
    public void loadMissingCloser_Fail() {
        TokenChain chain = readChain("(a");
        Context context = new Context(chain.get());
        CallGroup call = new CallGroup(context, chain.get());
        call.load();

        assertNotNull(call.getLineValue(), "Invalid line");

        assertErrors(context, Error.missingCloser);
    }

    @Test
    public void loadEmptyMissingCloser_Fail() {
        TokenChain chain = readChain("(");
        Context context = new Context(chain.get());
        CallGroup call = new CallGroup(context, chain.get());
        call.load();

        assertNull(call.getLineValue(), "Invalid line");

        assertErrors(context, Error.missingCloser, Error.lineEmptyLine);
    }
}

File: SimplexApp/simplex/parser/logic/line/call/CallGroup.test.js
const TokenChain = require("simplex/support/TokenChain.js");
const Context = require("simplex/parser/logic/Context.js");
const ContextSupport = require("simplex/support/ContextSupport.js");
const Error = require("simplex/parser/logic/error/Error.js");
const CallGroup= require("simplex/parser/logic/line/call/CallGroup.js");
const AssertError = require("simplex/support/AssertError.js");

const readChain = TokenChain.readChain;
const assertErrors = ContextSupport.assertErrors;
const assertNull = AssertError.assertNull;
const assertNotNull = AssertError.assertNotNull;

test('loadGroup', () => {
    let chain = readChain('(a)');
    let context = new Context(chain.get());
    let call = new CallGroup(context, chain.get());
    call.load();
    assertNotNull(call.getLineValue(), 'Invalid line');
    assertErrors(context);
});
test('loadEmpty_Fail', () => {
    let chain = readChain('()');
    let context = new Context(chain.get());
    let call = new CallGroup(context, chain.get());
    call.load();
    assertNull(call.getLineValue(), 'Invalid line');
    assertErrors(context, Error.lineEmptyLine);
});
test('loadMissingCloser_Fail', () => {
    let chain = readChain('(a');
    let context = new Context(chain.get());
    let call = new CallGroup(context, chain.get());
    call.load();
    assertNotNull(call.getLineValue(), 'Invalid line');
    assertErrors(context, Error.missingCloser);
});
test('loadEmptyMissingCloser_Fail', () => {
    let chain = readChain('(');
    let context = new Context(chain.get());
    let call = new CallGroup(context, chain.get());
    call.load();
    assertNull(call.getLineValue(), 'Invalid line');
    assertErrors(context, Error.missingCloser, Error.lineEmptyLine);
});